### PR TITLE
feat(upload): use LDS Cache

### PR DESCRIPTION
**Pull request by paulfouquet**
### Change Description:
`upload` command uses the LDS Cache by default.

### Notes for Testing:
Check the `README` for the new arguments

#### Source Code Documentation Tasks:
- [x] README updated (where applicable)
- [ ] CHANGELOG (Unreleased section) updated
- [ ] Docstrings / comments included to help explain code

#### User Documentation Tasks:
- [ ] Confluence updated (where applicable)

#### Testing Tasks:
- [ ] Added tests that fail without this change
- [x] All tests are passing in development environment
- [x] Reviewers assigned
- [ ] Linked to main issue for ZenHub board

### CODE REVIEW

File: topo_processor/metadata/metadata_loaders/metadata_loader_tiff.py [hunk 1]
```diff
@@ -27,6 +27,8 @@ def is_applicable(self, asset: Asset) -> bool:
     def load_metadata(self, asset: Asset) -> None:
 
         fs = get_fs(asset.source_path)
+        # FIXME: Should we download the file first as we could need it to do the coggification later?
```
Comment (blacha) [inline topo_processor/metadata/metadata_loaders/metadata_loader_tiff.py:30]: Reading a tiff header should be almost instant, It is generally one read from the head of the file.

Is rasterio reading a lot more than it should? Is there a way to force rasterio to only read the tiff headers?

File: topo_processor/metadata/metadata_loaders/metadata_loader_imagery_historic.py [hunk 2]
```diff
@@ -39,7 +42,11 @@ def is_applicable(self, asset: Asset) -> bool:
 
     def load_metadata(self, asset: Asset) -> None:
         if not self.is_init:
-            self.read_csv()
+            if not is_s3_path(asset.source_path):
```
Comment (blacha) [inline topo_processor/metadata/metadata_loaders/metadata_loader_imagery_historic.py:45]: what is the use case for using `asset.source_path` here? It should always just go to the `lds_cache`

File: topo_processor/metadata/metadata_loaders/metadata_loader_tiff.py (hunk 1 above)
Comment (paulfouquet) [reply topo_processor/metadata/metadata_loaders/metadata_loader_tiff.py:30]: I haven't investigate, but I can say it's quiet long!

Comment (blacha) [reply topo_processor/metadata/metadata_loaders/metadata_loader_tiff.py:30]: What is "quite long"?

File: topo_processor/metadata/metadata_loaders/metadata_loader_imagery_historic.py (hunk 2 above)
Comment (paulfouquet) [reply topo_processor/metadata/metadata_loaders/metadata_loader_imagery_historic.py:45]: Unit tests, should find another way but I was not inspired

Comment (blacha) [reply topo_processor/metadata/metadata_loaders/metadata_loader_imagery_historic.py:45]: could make `read_csv` take the file path then call the `.read_csv(asset_path)` in the unit test setup before processing files

File: topo_processor/metadata/metadata_loaders/metadata_loader_tiff.py (hunk 1 above)
Comment (paulfouquet) [reply topo_processor/metadata/metadata_loaders/metadata_loader_tiff.py:30]: 20-30 seconds

File: topo_processor/stac/lds_cache.py [hunk 3]
```diff
@@ -3,49 +3,44 @@
...
 
-        return version
+    return version
 
-    @staticmethod
-    def get_metadata_file_name(layer: str, version: str) -> str:
-        return layer + "_" + version + ".csv"
+
+def get_metadata_file_name(layer: str, version: str) -> str:
+    return layer + "_" + version + ".csv"
```
Comment (blacha) [inline topo_processor/stac/lds_cache.py:46]: not everything will be a `.csv`

File: topo_processor/metadata/metadata_loaders/metadata_loader_tiff.py (hunk 1 above)
Comment (blacha) [reply topo_processor/metadata/metadata_loaders/metadata_loader_tiff.py:30]: for how many files?

File: topo_processor/stac/lds_cache.py (hunk 3 above)
Comment (paulfouquet) [reply topo_processor/stac/lds_cache.py:46]: Is that not something we want to "normalize"?

File: topo_processor/metadata/metadata_loaders/metadata_loader_tiff.py (hunk 1 above)
Comment (paulfouquet) [reply topo_processor/metadata/metadata_loaders/metadata_loader_tiff.py:30]: Per file!

Comment (paulfouquet) [reply topo_processor/metadata/metadata_loaders/metadata_loader_tiff.py:30]:
```json
{"loader": "metadata.loader.imagery.tiff", "asset": "<topo_processor.stac.asset.Asset object at 0x7f13e2054d30>", "duration": 28361.980224609375, "level": 20, "time": 1638101651317, "v": 1, "pid": 30248, "msg": "Metadata Loaded"}
{"loader": "metadata.loader.imagery.historic", "asset": "<topo_processor.stac.asset.Asset object at 0x7f13e1feb0a0>", "duration": 1.566162109375, "level": 20, "time": 1638101651319, "v": 1, "pid": 30248, "msg": "Metadata Loaded"}
{"loader": "metadata.loader.imagery.tiff", "asset": "<topo_processor.stac.asset.Asset object at 0x7f13e1feb0a0>", "duration": 17520.66162109375, "level": 20, "time": 1638101668840, "v": 1, "pid": 30248, "msg": "Metadata Loaded"}
{"loader": "metadata.loader.imagery.historic", "asset": "<topo_processor.stac.asset.Asset object at 0x7f13e2054730>", "duration": 4.18115234375, "level": 20, "time": 1638101668845, "v": 1, "pid": 30248, "msg": "Metadata Loaded"}
{"loader": "metadata.loader.imagery.tiff", "asset": "<topo_processor.stac.asset.Asset object at 0x7f13e2054730>", "duration": 22965.641845703125, "level": 20, "time": 1638101691811, "v": 1, "pid": 30248, "msg": "Metadata Loaded"}
{"loader": "metadata.loader.imagery.historic", "asset": "<topo_processor.stac.asset.Asset object at 0x7f13e2054070>", "duration": 2.7060546875, "level": 20, "time": 1638101691815, "v": 1, "pid": 30248, "msg": "Metadata Loaded"}
{"loader": "metadata.loader.imagery.tiff", "asset": "<topo_processor.stac.asset.Asset object at 0x7f13e2054070>", "duration": 24532.73388671875, "level": 20, "time": 1638101716348, "v": 1, "pid": 30248, "msg": "Metadata Loaded"}
{"loader": "metadata.loader.imagery.historic", "asset": "<topo_processor.stac.asset.Asset object at 0x7f13e2054be0>", "duration": 1.12060546875, "level": 20, "time": 1638101716349, "v": 1, "pid": 30248, "msg": "Metadata Loaded"}
{"loader": "metadata.loader.imagery.tiff", "asset": "<topo_processor.stac.asset.Asset object at 0x7f13e2054be0>", "duration": 25956.140625, "level": 20, "time": 1638101742305, "v": 1, "pid": 30248, "msg": "Metadata Loaded"}
{"loader": "metadata.loader.imagery.historic", "asset": "<topo_processor.stac.asset.Asset object at 0x7f13e2054d90>", "duration": 1.721923828125, "level": 20, "time": 1638101742307, "v": 1, "pid": 30248, "msg": "Metadata Loaded"}
{"loader": "metadata.loader.imagery.tiff", "asset": "<topo_processor.stac.asset.Asset object at 0x7f13e2054d90>", "duration": 21477.30859375, "level": 20, "time": 1638101763785, "v": 1, "pid": 30248, "msg": "Metadata Loaded"}
{"loader": "metadata.loader.imagery.historic", "asset": "<topo_processor.stac.asset.Asset object at 0x7f13e2054280>", "duration": 4.956787109375, "level": 20, "time": 1638101763791, "v": 1, "pid": 30248, "msg": "Metadata Loaded"}
{"loader": "metadata.loader.imagery.tiff", "asset": "<topo_processor.stac.asset.Asset object at 0x7f13e2054280>", "duration": 25951.318115234375, "level": 20, "time": 1638101789743, "v": 1, "pid": 30248, "msg": "Metadata Loaded"}
```
Where `duration` is in `ms`

Comment (blacha) [reply topo_processor/metadata/metadata_loaders/metadata_loader_tiff.py:30]: That is really really slow :( Basemaps's Reads EPSG/file size from a 1000 files in few seconds... wonder why this is 1000x slower

File: topo_processor/metadata/metadata_loaders/metadata_loader_imagery_historic.py (hunk 2 above)
Comment (paulfouquet) [reply topo_processor/metadata/metadata_loaders/metadata_loader_imagery_historic.py:45]: Will go along with https://github.com/linz/topo-processor/pull/457

**Review by amfage — APPROVED** [review]

**Review by MDavidson17 — APPROVED** [review]

**Pull request merged**